src/NuGet.Core/NuGet.Frameworks/DefaultCompatibilityListProvider.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Linq;
+
+namespace NuGet.Frameworks
+{
+    public sealed class DefaultCompatibilityListProvider : IFrameworkCompatibilityListProvider
+    {
+        private readonly IFrameworkNameProvider _nameProvider;
+        private readonly IFrameworkCompatibilityProvider _compatibilityProvider;
+        private readonly FrameworkReducer _reducer;
+
+        public DefaultCompatibilityListProvider()
+            : this(DefaultFrameworkNameProvider.Instance, DefaultCompatibilityProvider.Instance)
+        {
+        }
+
+        public DefaultCompatibilityListProvider(IFrameworkNameProvider nameProvider, IFrameworkCompatibilityProvider compatibilityProvider)
+        {
+            _nameProvider = nameProvider;
+            _compatibilityProvider = compatibilityProvider;
+            _reducer = new FrameworkReducer(_nameProvider, _compatibilityProvider);
+        }
+
+        public IEnumerable<NuGetFramework> GetFrameworksSupporting(NuGetFramework target)
+        {
+            var remaining = _nameProvider
```

Comment:
We could potentially have memoization for this method (in addition to what we have in `IFrameworkCompatibilityProvider.IsCompatible`.